csharp/src/Ice/Reference.cs

```diff
@@ -236,16 +236,16 @@ public override string ToString()
                 s.Append(" -s");
             }
 
-            if (!_protocol.Equals(Ice.Util.Protocol_1_0))
+            s.Append(" -p ");
+            if (Communicator.ToStringMode >= ToStringMode.Compat)
             {
-                //
-                // We only print the protocol if it's not 1.0. It's fine as
-                // long as we don't add Ice.Default.Protocol, a
-                // stringified proxy will convert back to the same proxy with
-                // stringToProxy.
-                //
-                s.Append(" -p ");
-                s.Append(Ice.Util.ProtocolToString(_protocol));
+                // x.0 style
+                s.Append(_protocol.ToString("D"));
+                s.Append(".0");
+            }
+            else
+            {
+                s.Append(_protocol.ToString().ToLower());
```

Comment:
A `ToString(ToStringMode mode)` method defined on a `Protocol` struct would be better.

Reply:
An enum is better for Protocol and it's not possible to override an enum's ToString. You can provide a ToString extension method but that's confusing.